app/domain/authentication/authn_k8s/kubectl_exec.rb

```diff
@@ -181,27 +200,44 @@ def copy(k8s_object_lookup:,
         container: 'authenticator')
         execute(
           k8s_object_lookup: k8s_object_lookup,
-          pod_namespace: pod_namespace,
-          pod_name: pod_name,
-          container: container,
-          cmds: ['tar', 'xvf', '-', '-C', '/'],
-          body: tar_file_as_string(path, content, mode),
-          stdin: true
+          pod_namespace:     pod_namespace,
+          pod_name:          pod_name,
+          container:         container,
+          cmds:              %w(sh),
+          body:              copy_script(path, content, mode),
+          stdin:             true
         )
       end
 
       private
 
-      def tar_file_as_string(path, content, mode)
-        tarfile = StringIO.new("")
-
-        Gem::Package::TarWriter.new(tarfile) do |tar|
-          tar.add_file(path, mode) do |tf|
-            tf.write(content)
-          end
-        end
-
-        tarfile.string
+      def copy_script(path, content, mode)
+        parts = path.split('/')
+        filename = parts.pop
+        dir = parts.join('/')
+
+        # We first copy the content into a temporary file and only then move it to
+        # the desired path as the client polls on its existence and we want it to
+        # exist only when the whole content is present
+        tmp_cert = "#{dir}/tmp_#{filename}"
+
+        # We redirect the output also to a log file on the authn-client container
+        # for debugging purposes.
+        "
+#!/bin/sh
+
+copy_file() {
+  cat > #{tmp_cert} <<EOF
+#{content}
+EOF
+  chmod #{mode} #{tmp_cert}
+  mv #{tmp_cert} #{path}
+}
+
+copy_file 2>&1 | tee -a #{COPY_FILE_LOG_FILE}
```

Comment:
I think less dependencies is better.
Redirection does not require `tee`, you can simply call `> logfile`.
See https://stackoverflow.com/a/315113

Reply:
`tee` is GNU coreutils so we probably won't have a problem with that but I think he intended to print both to stdout _and_ the file.

Reply:
the `stdout` is good for writing the logs in the server, but the server closes the socket immediately and doesn't wait for any stdout.

Reply:
yes i changed it to
```
set_file_content > \"#{LOG_FILE}\" 2>&1
```
as we don't need to return stdout in the socket (as it is closed)